Failure to deploy specs website due to mismatch between katex and mdbook version

eg https://github.com/anoma/specs/runs/7212960732?check_suite_focus=true
```
2022-07-06 10:40:55 [INFO] (mdbook::book): Book building has started
Error: Katex preprocessor/renderer using different mdbook version, 0.4.18,than it was built against, 0.4.19
Error: -0[6](https://github.com/anoma/specs/runs/7234853079?check_suite_focus=true#step:6:7) 10:40:55 [ERROR] (mdbook::utils): Error: The "katex" preprocessor exited unsuccessfully with exit status: 1 status
make: *** [Makefile:[7](https://github.com/anoma/specs/runs/7234853079?check_suite_focus=true#step:6:8): build] Error 101
Error: Process completed with exit code 2.
```

--- a/.github/workflows/build-pr.yml
+++ b/.github/workflows/build-pr.yml
@@ -3,7 +3,7 @@ name: Build PR
 on:
   pull_request:
     branches:
-      - master
+      - main
 
 jobs:
   build:

--- a/.github/workflows/build-pr.yml
+++ b/.github/workflows/build-pr.yml
@@ -24,7 +24,7 @@ jobs:
           ~/.cargo
         key: ${{ hashFiles('Makefile') }}
     - name: Install cargo deps
-      run: make dev-deps
+      run: make ci-deps
     - name: Generate website
       run: make build
     - name: Archive rendered site

--- a/.github/workflows/deploy-master.yml
+++ b/.github/workflows/deploy-master.yml
@@ -24,7 +24,7 @@ jobs:
           ~/.cargo
         key: ${{ hashFiles('Makefile') }}
     - name: Install cargo deps
-      run: make dev-deps
+      run: make ci-deps
     - name: Generate website
       run: make build
     - name: Create tag folder
@@ -33,6 +33,6 @@ jobs:
     - name: Deploy
       uses: JamesIves/github-pages-deploy-action@v4.2.3
       with:
-        branch: gh-pages 
+        branch: gh-pages
         folder: docs
         clean: false

--- a/.github/workflows/deploy-master.yml
+++ b/.github/workflows/deploy-master.yml
@@ -3,7 +3,7 @@ name: Deploy Master
 on:
   push:
     branches:
-      - master
+      - main
 
 jobs:
   build-deploy:
@@ -29,7 +29,7 @@ jobs:
       run: make build
     - name: Create tag folder
       run: |
-        mkdir -p docs/master && mv book/html/* docs/master
+        mkdir -p docs/main && mv book/html/* docs/main
     - name: Deploy
       uses: JamesIves/github-pages-deploy-action@v4.2.3
       with:

--- a/.github/workflows/deploy-tag.yml
+++ b/.github/workflows/deploy-tag.yml
@@ -24,7 +24,7 @@ jobs:
           ~/.cargo
         key: ${{ hashFiles('Makefile') }}
     - name: Install cargo deps
-      run: make dev-deps
+      run: make ci-deps
     - name: Generate website
       run: make build
     - uses: olegtarasov/get-tag@v2.1
@@ -32,7 +32,7 @@ jobs:
       with:
         tagRegex: "v(.*)"
     - name: Create tag folder
-      env: 
+      env:
         TAG_NAME: ${{ steps.tagName.outputs.tag }}
       run: |
         mkdir -p docs/$TAG_NAME && mv book/html/* docs/$TAG_NAME

--- a/Makefile
+++ b/Makefile
@@ -6,10 +6,17 @@ serve:
 build:
 	mdbook build
 
+# these versions are guaranteed to be able to build the specs mdbook
+ci-deps:
+	$(cargo) install mdbook --version 0.4.19
+	$(cargo) install mdbook-mermaid --version 0.11.0
+	$(cargo) install mdbook-linkcheck --version 0.7.6
+	$(cargo) install --git https://github.com/heliaxdev/mdbook-katex.git --rev 2b37a542808a0b3cc8e799851514e145990f1e3a
+
 dev-deps:
 	$(cargo) install mdbook
 	$(cargo) install mdbook-mermaid
 	$(cargo) install mdbook-linkcheck
-	$(cargo) install --git https://github.com/heliaxdev/mdbook-katex.git --rev 2b37a542808a0b3cc8e799851514e145990f1e3a
+	$(cargo) install --git https://github.com/heliaxdev/mdbook-katex.git
 
 .PHONY: build serve dev-deps

